### DISCUSSION TITLE
sketch and minimal themes

### DISCUSSION THREAD

**Discussion by davidjgraph** [jgraph/drawio#1924] [Ideas]
We've a couple of themes we'd like some feedback on, [minimal](https://app.diagrams.net/?splash=0&ui=min) and [sketch](https://sketch.diagrams.net).

The minimal theme is used when the screensize is below a certain limit to reduce the number of menus on mobile screens.

The sketch theme looks more like a whiteboarding app, with the rough.js sketch style enabled by default.

The question is whether either (probably more sketch) would be useful as an easy to access setting in either diagrams.net or draw.io for Confluence/Jira cloud. Is it something you could see be more accessible for less technical users? Any thought on the UX flow that would allow users to access it?

**Comment by bitsdance** [comment]
I use confluence/jira, teams and draw.io a lot to create solutions, architecture etc.

On a call we use teams whiteboard generally
On confluence we use gliffy 

but I am impressed by sketch as it has potential to replace teams whiteboard but let me try more and I can leave review. however you can use my use cases for further ideas. one thing which i'd love to see is version control the source file like XML which is not really an XML

**Comment by dannybrian** [comment]
Love minimal. It would be my default.

**Reply by davidjgraph** [reply]
Select extras->theme->minimal in the menu and that browser will remember to use that theme.

**Reply by Xgz20** [reply]
very very good

**Reply by chipbite** [reply]
But extras -> theme -> minimal is not available when starting the editor from confluence, right?
I am not seeing it, atleast. Config?
<img src="https://user-images.githubusercontent.com/559199/114850689-55588880-9de1-11eb-9bea-fb04ad9ad298.png" width=300/>

**Reply by davidjgraph** [reply]
No, only the default Atlassian blue theme is available in Confluence Cloud. Confluence has the problem of needing additional testing to ensure everything fits in with the host product, allowing all the theme would add a considerable testing load. In Confluence the idea would be more like having the sketch theme as a specific "draw.io sketch" macro, something along those lines.

If there are specific thing in minimal that would be useful, like making the main panels (un)dockable, we can look at those as specific issues.

**Reply by chipbite** [reply]
Aha, thanks @davidjgraph! I understand. 
I wrote my thoughts below - the experience I am after is max real estate so skipping the toolbar is great, as few elements visible as possible, panels visible when I need them (and thus toggle:able).

**Reply by davidjgraph** [reply]
Added as https://github.com/jgraph/drawio/issues/1930

**Comment by internetscooter** [comment]
Love the them - general feedback on sketch... yes, I know this is beyond what you are asking but I will state it anyway...

- I think the Freehand Start/Stop widget is redundant. You could get away with just highlighting the pen icon when it is active (maybe even have a pen cursor icon when active).
- Sketch line should ideally be in sketch style when drawing (nice to have but not essential)

This is great stuff, I would love an easy way to make these themes myself, share and tweak them (however I know this would take a large amount of work to refactor code to do this)

**Reply by chipbite** [reply]
Despite my penchant for minimal and hideable UIs, I think the freehand panel adds needed clarity, especially if you are not used to the "mode" aspect of it. 

I could (of course) see that you could implement what @internetscooter is proposing on the pen icon, in addition to the freehand panel, and make the freehand panel closeable.

**Reply by davidjgraph** [reply]
What we're trying to work out if whether users in sketch tend to just freehand most of the time (i.e. literally sketch) as part of a collaborative meeting. If they do the explicit mode is better. We just need to work out how we're going to work that out...

**Comment by ghost** [comment]
### Needs

I have a real love for the whiteboard but these days we work remotly 90% of time. I have to schematize otherwise. 

When I want to draw something simple, quick but still clean during a Teams call, I'm a bit stuck :
- Classic version of drawio is not simple enough ;
- Paint is not clean and reusable enough ;
- Teams whiteboard is limited too (and I don't have access) ;
- Gimp, no thanks.

### Sketch feedbacks

I really appreciate the `sketch` version that seems to meet my needs.

- I agree with @internetscooter, the `Freehand` widget seems unnecessary to me : left clic on the tool to draw, right clic to leave. Currently, its use is not intuitive for me ;
- I would like the `pencil` to be the first tool, above "text" ;
- A Greenshot feature I would like to have is `the counter`. It's really usefull to numbered to explain a path to follow or the data life cycle for example. It could be simply '1' in a circle, select it in the menu, then when you apply it somewhere, with the left clic, it goes to '2', etc... Clic right to quit the numbering, clic left on the tool or in a number to resume.

Thank you for beautifying my working days,

Regards, Guillaume

ps: please save the exporting path...

**Comment by easa** [comment]
May we have dark-minimal?
Also, the movability of the toolbox and other boxes is not necessary.

**Reply by chipbite** [reply]
I disagree, I like that the panels are movable. For some panels (actually layers), it is essential, since it depends on the view and diagram where you prefer to have them. For the larger panels, I kindof agree, not as important. But resizing and moving them is valuable, IMO.

**Reply by easa** [reply]
I think you are right.

**Comment by Niwook** [comment]
I think each theme has its utility. Both are very interesting for me. Really a great work.

**Comment by chipbite** [comment]
OH YES! This is great! Very promising indeed! Actually, **Awesome** is the word! ❤❤❤🎉

----
⚠❔ One thing above all others: What is the **keyboard shortcut to toggle the shapes panel?** 
With that keyboard shortcut, the UI would start to move into perfection.
----
Some other comments after some testing:
* Second big thing - Connector type setting - Lack (both min and sketch): If not sketching I am diagramming boxes and arrows. **One of the most common operations is changing the connector type** from right angle to straight to vertical to horizontal, etc. Even using full UI, this is not the swiftest operation, considering how often you do it. Now, in these two - min & sketch - I have to open or have the format panel open, and then use the small dropdown. IMO - for all modes - if I rightclick on a connector the very first option in the context menu should be to switch connector type, and there ideally should be a keyboard shortcut to quickly move between them, popping up a small navigator (the existing menu widget would work great) to show which one I am at.
* Good to have: In sketch, I would like to hide the custom small sketch shapes panel. Via keyboard.
* Observation: In sketch, I can open a multipage file, and switch pages using keyboard, but I get no visual feedback at all. I definitely do not want the tabs back, but there could be other ways to alert the user of existing tabs and the you switch, akin to alt-tabbing in windows environments (but definitely skip the previews).
 

I think both are great and work as you intended. Minimal or even sketch - if the sketch panel would be hideable and with page/tab switching feedback - would be my new default mode. 

Love how I get more screen real estate and peace of mind, **zen diagramming**! 🧘‍♂️

(Just love this! **Huge benefit**: We _can_ now toggle the shapes sidebar - no more sticky-dragging it sideways! HUGE improvement!)

**Reply by chipbite** [reply]
OK! So now I accidentally found out I have ctrl-shift-j to toggle shapes panel (atleast in min ui). 
Great!

Is that coming to confluence as well?

**Reply by davidjgraph** [reply]
Is what coming, the min UI?

**Reply by chipbite** [reply]
Hi!
> Is what coming, the min UI?

**ctrl-shift-j** to _toggle shapes panel_ (to confluence)!

Really valuable! 😊

**Reply by chipbite** [reply]
Update: ctrl-shift-k is (now) the shortcut to toggle shapes panel. 
( Note: The shortcut seems to be only available in minimal, I miss it in standard theme and in confluence. )

**Reply by chipbite** [reply]
I created an issue to enable this keyboard shortcut to toggle shapes panel in the confluence editor also: #2684

**Comment by andersonnc** [comment]
I like sketching a lot. Missing the option to open a new tab, here it is visible only if you have 2 tabs in a single file.
![image](https://user-images.githubusercontent.com/31364907/115098215-220f1a00-9f05-11eb-8afc-be1974449094.png)

**Reply by davidjgraph** [reply]
The aim of the sketch UI is to try to remove as much as possible. It's also really meant to be an infinite white-board style of app. The page functionality is purposefully removed as part of that.

What I'm keen to avoid is stripping the UI down and then adding everything back in one by one. 

So, thank you for the feedback, currently I would suggest to use one of the other, larger, UIs for that functiohality.

**Comment by OliEfr** [comment]
Please just dont try to copy miro!

**Reply by davidjgraph** [reply]
I'm not really sure what you mean here.

**Comment by vianneychevalier** [comment]
Great work. 
I often work with less technical users (consulting interviews, process mapping, etc.). I see myself proposing the sketch theme. I would change the default colors accessible in Format\Style when using this theme: the colors on the first colors are very faint on low quality screens or when projected, which are often found in the same context as "low-tech" users. 

I understand their use for text contrast, maybe increasing the default contour line thickness would solve it?

![image](https://user-images.githubusercontent.com/18335583/115226642-c664ac80-a10f-11eb-8b25-05ff8616c8ec.png)

One more thing: in combination with the freehand drawing, could we get a "fill" functionality? I may have not discovered it yet...

**Comment by alderg** [comment]
Thanks for all your feedback! We've implemented most of these ideas in version 14.6.6. Visit https://app.diagrams.net/clear.html and then https://sketch.diagrams.net to get the update.

**Comment by raydeo** [comment]
Hi, 

Excellent work with the new themes!

Please can you tell me where I can set the default theme in code?
I would like the app to start with Minimal theme.

Many thanks in advance for your reply!

**Comment by davidjgraph** [comment]
https://www.diagrams.net/blog/drawio-sketch-macro

**Reply by borekb** [reply]
I'm loving the sketch mode! It makes it feel like Excalidraw but with all the advantages of diagrams.net. Great job!
